$NetBSD: patch-gio_tests_meson.build,v 1.2 2020/01/23 23:54:22 sevan Exp $

Darwin: Set LD_LIBRARY_PATH to run uninstalled executable with @rpath.
Check objcopy(1) supports --add-symbol which may not be available on
legacy systems.
https://gitlab.gnome.org/GNOME/glib/commit/dac4a48d14461a118d8ad7825a39e224bf128807

--- gio/tests/meson.build.orig	2019-12-19 16:33:15.000000000 +0000
+++ gio/tests/meson.build
@@ -470,7 +470,9 @@ if not meson.is_cross_build() or meson.h
   plugin_resources_c = custom_target('plugin-resources.c',
     input : 'test4.gresource.xml',
     output : 'plugin-resources.c',
-    command : [glib_compile_resources,
+    command : [env_program.path(),
+               'LD_LIBRARY_PATH=gio:glib:gmodule:gobject',
+    	       glib_compile_resources,
                '--target=@OUTPUT@',
                '--sourcedir=' + meson.current_source_dir(),
                '--generate-source',
@@ -494,7 +496,9 @@ if not meson.is_cross_build() or meson.h
   test_gresource = custom_target('test.gresource',
     input : 'test.gresource.xml',
     output : 'test.gresource',
-    command : [glib_compile_resources,
+    command : [env_program.path(),
+               'LD_LIBRARY_PATH=gio:glib:gmodule:gobject',
+    	       glib_compile_resources,
                '--target=@OUTPUT@',
                '--sourcedir=' + meson.current_source_dir(),
                '--sourcedir=' + meson.current_build_dir(),
@@ -505,7 +509,9 @@ if not meson.is_cross_build() or meson.h
   test_resources2_c = custom_target('test_resources2.c',
     input : 'test3.gresource.xml',
     output : 'test_resources2.c',
-    command : [glib_compile_resources,
+    command : [env_program.path(),
+               'LD_LIBRARY_PATH=gio:glib:gmodule:gobject',
+    	       glib_compile_resources,
                '--target=@OUTPUT@',
                '--sourcedir=' + meson.current_source_dir(),
                '--generate',
@@ -516,7 +522,9 @@ if not meson.is_cross_build() or meson.h
   test_resources2_h = custom_target('test_resources2.h',
     input : 'test3.gresource.xml',
     output : 'test_resources2.h',
-    command : [glib_compile_resources,
+    command : [env_program.path(),
+               'LD_LIBRARY_PATH=gio:glib:gmodule:gobject',
+    	       glib_compile_resources,
                '--target=@OUTPUT@',
                '--sourcedir=' + meson.current_source_dir(),
                '--generate',
@@ -528,7 +536,9 @@ if not meson.is_cross_build() or meson.h
     input : 'test2.gresource.xml',
     depends : big_test_resource,
     output : 'test_resources.c',
-    command : [glib_compile_resources,
+    command : [env_program.path(),
+               'LD_LIBRARY_PATH=gio:glib:gmodule:gobject',
+    	       glib_compile_resources,
                '--target=@OUTPUT@',
                '--sourcedir=' + meson.current_source_dir(),
                '--sourcedir=' + meson.current_build_dir(),
@@ -539,7 +549,9 @@ if not meson.is_cross_build() or meson.h
   digit_test_resources_c = custom_target('digit_test_resources.c',
     input : '111_digit_test.gresource.xml',
     output : 'digit_test_resources.c',
-    command : [glib_compile_resources,
+    command : [env_program.path(),
+               'LD_LIBRARY_PATH=gio:glib:gmodule:gobject',
+    	       glib_compile_resources,
                '--target=@OUTPUT@',
                '--sourcedir=' + meson.current_source_dir(),
                '--sourcedir=' + meson.current_build_dir(),
@@ -550,7 +562,9 @@ if not meson.is_cross_build() or meson.h
   digit_test_resources_h = custom_target('digit_test_resources.h',
     input : '111_digit_test.gresource.xml',
     output : 'digit_test_resources.h',
-    command : [glib_compile_resources,
+    command : [env_program.path(),
+               'LD_LIBRARY_PATH=gio:glib:gmodule:gobject',
+    	       glib_compile_resources,
                '--target=@OUTPUT@',
                '--sourcedir=' + meson.current_source_dir(),
                '--generate',
@@ -577,9 +591,13 @@ if not meson.is_cross_build() or meson.h
   # Support for --add-symbol was added to LLVM objcopy in 2019
   # (https://reviews.llvm.org/D58234). FIXME: This test could be enabled for
   # LLVM once that support is in a stable release.
+  objcopy_valid = false
   objcopy = find_program('objcopy', required : false)
+  if objcopy.found()
+      objcopy_valid = run_command(objcopy, '--help').stdout().contains('--add-symbol')
+  endif
 
-  if build_machine.system() == 'linux' and cc.get_id() == 'gcc' and objcopy.found()
+  if build_machine.system() == 'linux' and cc.get_id() == 'gcc' and objcopy.found() and objcopy_valid
     test_gresource_binary = custom_target('test5.gresource',
       input : 'test5.gresource.xml',
       output : 'test5.gresource',
